core/utils/metadata-utils/src/main/java/datawave/query/model/DateFrequencyMap.java

```diff
@@ -0,0 +1,341 @@
+     * @return the count
+     */
+    public Frequency get(String date) {
+        return dateToFrequencies.get(date);
+    }
+
+    /**
+     * Return whether this map contains a mapping for the given date.
+     *
+     * @param date
+     *            the date
+     * @return true if a mapping exists for the given date, or false otherwise
+     */
+    public boolean contains(String date) {
+        return dateToFrequencies.containsKey(date);
+    }
+
+    /**
+     * Clear all mappings in this {@link DateFrequencyMap}.
+     */
+    public void clear() {
+        this.dateToFrequencies.clear();
+    }
+
+    @Override
+    public Set<String> keySet() {
+        return dateToFrequencies.keySet();
+    }
+
+    @Override
+    public Collection<Frequency> values() {
+        return dateToFrequencies.values();
+    }
+
+    /**
+     * Returns a {@link Set} view of the mappings contained within this map, sorted in ascending by order.
+     *
+     * @return a {@link Set} view of the mappings
+     */
+    public Set<Map.Entry<String,Frequency>> entrySet() {
+        return this.dateToFrequencies.entrySet();
+    }
+
+    /**
+     * Returns a view of the portion of this {@link DateFrequencyMap}'s underlying map whose keys range from startDate (inclusive) to endDate (inclusive).
+     *
+     * @param startDate
+     *            the start date
+     * @param endDate
+     *            the end date
+     * @return the map view
+     */
+    public SortedMap<String,Frequency> subMap(String startDate, String endDate) {
+        return dateToFrequencies.subMap(startDate, true, endDate, true);
+    }
+
+    /**
+     * Returns the earliest date in this {@link DateFrequencyMap}.
+     *
+     * @return the earliest date
+     */
+    public String earliestDate() {
+        return dateToFrequencies.firstKey();
+    }
+
+    @Override
+    public void write(DataOutput dataOutput) throws IOException {
+        // Write the version
+        WritableUtils.writeEnum(dataOutput, VERSION.DFM1);
+
+        // Write the format type depending on the data size
+        // we are guessing that we break even somewhere around 1K
+        FORMAT format = FORMAT.BASIC;
+        if (size() > 56) {
+            format = FORMAT.GZIP;
+        }
+        WritableUtils.writeEnum(dataOutput, format);
+
+        switch (format) {
+            case BASIC:
+                writeData(dataOutput);
+                break;
+            case GZIP:
+                // create a gzipped output
+                ByteArrayOutputStream output = new ByteArrayOutputStream();
+                try (DataOutputStream out = new DataOutputStream(output)) {
+                    writeData(out);
+                }
+                output.close();
+                WritableUtils.writeCompressedByteArray(dataOutput, output.toByteArray());
+                break;
+            default:
+                throw new UnsupportedEncodingException("Unknown output format " + DEFAULT_FORMAT);
```

Comment:
I think this should just be "format", otherwise the DEFAULT_FORMAT will mask whatever value was actually written.